src/components/diff-view.jsx

```diff
@@ -58,17 +57,19 @@ export default class DiffView extends React.Component {
       );
     }
 
-    if (!diff) {
+    if (!this.state.diffData) {
       return <Loading />;
     }
-    else if (diff instanceof Error) {
+    else if (this.state.diffData instanceof Error) {
       return (
         <p className="alert alert-danger" role="alert">
-          Error: {diff.message}
+          Error: {this.state.diffData.message}
         </p>
       );
     }
 
+    const diff = this.state.diffData.diff;
```

Comment:
Aren’t the different views going to potentially need the metadata that will be directly in `diffData` (not just what’s in `diffData.diff`) in order to render properly? e.g. in #156, the individual diff views are in charge of what to display based on whether there were changes. The new metadata to make that easier will be in `diffData` here.

Reply:
Yes, but as mentioned
> I thought it made more sense to pass just the diff to the different views rather than the whole object because it's all they need. In the future, to fix #129 (#156 PR) we can pass along `change_count` as a new prop.

I think it makes sense that `diff-view` is the only thing that needs to know how the metadata is structured and passes along what is needed to the diff views as props. That way if we ever re-structure the metadata again or if someone else reuses these components but without our structure, it would be more maintainable. 

But, I'm still new to React, so if my reasoning is off, please let me know.

Reply:
Well, some diffs have more metadata—e.g. `html_token` has `deletion_count` and `insertion_count` to go along with `change_count`. It also carries other metadata in the `<head>` of the combined diff that we should probably extract out (like a diff of the page title) so it can be more easily displayed. Having a clear contract with props is nice, but I fear it’s going to get complicated as we add useful metadata to various diffs.

Reply:
Ah, in that case, will change.

Reply:
(Otherwise I would totally agree! This made the interface between components way more clear.)

Reply:
Yea, I'm learning a lot about weighing practical considerations vs ideal situations. It's hard not to try to do everything strictly "by the book", so it's to good to learn to bend and be flexible.